src/db/pool.cr

```diff
@@ -47,6 +60,56 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T, delay : Float64)
+      return unless connection.responds_to?(:check)
+
+      sleep delay
+
+      sync do
+        connection_idle = @idle.includes? connection
+        return unless connection_idle # let the next reaping attempt get it
+
+        delete connection
+      end
+
+      connection.check
+
+      sync do
+        @total << connection
+        @idle << connection
+      end
+    rescue exc : DB::ConnectionLost
+      connection.close
+
+      sync do
+        add_resource! if can_increase_pool?
+      end
+    end
+
+    def add_resource!
+      @inflight += 1
+      r = unsync { build_resource }
+    ensure # prevent inflight from incrementing if build_resource fails
+      @inflight -= 1
+      r
```

Comment:
Caching the return value is unnecessary. The method returns the value of the main body, not the `ensure` block.
```suggestion
      unsync { build_resource }
    ensure # prevent inflight from incrementing if build_resource fails
      @inflight -= 1
```